cpp/src/arrow/record_batch.cc

```diff
@@ -247,6 +248,97 @@ Result<std::shared_ptr<StructArray>> RecordBatch::ToStructArray() const {
                                        /*offset=*/0);
 }
 
+template <typename DataType>
+inline void ConvertColumnsToTensor(const RecordBatch& batch, uint8_t* out) {
+  using CType = typename arrow::TypeTraits<DataType>::CType;
+  auto* out_values = reinterpret_cast<CType*>(out);
+
+  // Loop through all of the columns
+  for (int i = 0; i < batch.num_columns(); ++i) {
+    const auto* in_values = batch.column(i)->data()->GetValues<CType>(1);
```

Comment:
I was looking into adding a check but am not sure how to and not even sure if it is necessary.

What would be good to check is that the `in_values` are casted to `CType` so that the resulting `out_values` are copied from the correct type. But that should be clear or am I missing something?

Reply:
This is templated and so both `out_values` and `in_values` are typed to the same `CType` template variable, so doing an additional DCHECK for that indeed seems unnecessary